opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -220,6 +226,24 @@ def on_end(self, span: ReadableSpan) -> None:
             with self.condition:
                 self.condition.notify()
 
+    def _at_fork_reinit(self):
+        # could be in an inconsistent state after fork, reinitialise by calling `_at_fork_reinit`
+        # (creates a new lock internally https://github.com/python/cpython/blob/main/Python/thread_pthread.h#L727)
+        # if exists, otherwise create a new one.
+        if hasattr(self.condition, "_at_fork_reinit"):
+            self.condition._at_fork_reinit()
+        else:
+            self.condition = threading.Condition(threading.Lock())
+
+        self.queue.clear()
+
+        # worker_thread is local to a process, only the thread that issued fork continues
+        # to exist. A new worker thread must be started in child process.
+        self.worker_thread = threading.Thread(
```

Comment:
@lonewolf3739 
This is great. Is this what you mean by making the processor "fork-safe"? We've had some complaints from guicorn users not getting telemetry because the worker thread is not copied.